### PR TITLE
fix: fixes multi-platform push

### PR DESCRIPTION
**Pull request by bruce-szalwinski-he**
Addresses #2.  

Thanks for the awesome plugin.  I fixed the push issue by joining the platforms.  While I was here, I added a few things:

- unit tests
- pre-commit
- bumped a few dependencies to get tests and typing working

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+repos:
```
Comment (bruce-szalwinski-he) [inline .pre-commit-config.yaml:1]: do some checks before committing.

File: Makefile [hunk 2]
```diff
@@ -26,8 +26,10 @@ format:
 .PHONY: compile
 compile: format
 	@poetry check
-	@poetry run ruff poetry_docker_plugin
-	@poetry run mypy poetry_docker_plugin
+	@poetry run ruff check --diff --no-fix .
```
Comment (bruce-szalwinski-he) [inline Makefile:29]: check format and run tests

File: poetry_docker_plugin/docker_builder.py [hunk 3]
```diff
@@ -283,80 +283,21 @@ def build(
...
-                    *[arg for tag in image_tags for arg in ["--tag", tag]],
-                    "--file",
-                    f"dist/{dockerfile_name}",
-                    os.path.abspath("dist"),
-                ],
-                stdin=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                universal_newlines=True,
-            )
+        build_command = BuildCommand(image_tags, platform, arguments, dockerfile_name)
```
Comment (bruce-szalwinski-he) [inline poetry_docker_plugin/docker_builder.py:286]: created a class to build the command to make it easy to test.  didn't want to mock `subprocess.run`.

File: poetry_docker_plugin/docker_builder.py [hunk 4]
```diff
@@ -283,80 +283,21 @@ def build(
...
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            universal_newlines=True,
+        )
 
         if result.returncode == 0:
             self._io.write_line("<info>[INFO]:</info> Image tags successfully created!")
 
         if push and len(platform) > 1:
+            push_command = PushCommand(image_tags, platform, arguments, dockerfile_name)
```
Comment (bruce-szalwinski-he) [inline poetry_docker_plugin/docker_builder.py:298]: same thing.  created a class to create the push command to make it easy to test the command was being created.

File: poetry_docker_plugin/docker_builder.py [hunk 5]
```diff
@@ -381,3 +322,89 @@ def __push(self, image_tag: str) -> None:
...
+        self.dockerfile_name = dockerfile_name
+        self.platform = platform
+
+    def command(self) -> List[str]:
+        return [
+            "docker",
+            "buildx",
+            "build",
+            "--push",
+            f"--platform={','.join(self.platform)}",
```
Comment (bruce-szalwinski-he) [inline poetry_docker_plugin/docker_builder.py:401]: This is the key line of the entire PR :).

File: pyproject.toml [hunk 6]
```diff
@@ -108,6 +108,7 @@ import_heading_firstparty="Project"
 [tool.ruff]
 line-length = 120
 target-version = "py38"
+src = ["poetry_docker_plugin", "tests"]
```
Comment (vagmcs) [inline pyproject.toml:111]: This should be removed.

**Review by vagmcs — COMMENTED** [review]
Thank you for fixing the issue. I just added a couple of minor comments regarding the Makefile.

File: Makefile [hunk 7]
```diff
@@ -26,8 +26,10 @@ format:
 .PHONY: compile
 compile: format
 	@poetry check
-	@poetry run ruff poetry_docker_plugin
-	@poetry run mypy poetry_docker_plugin
+	@poetry run ruff check --diff --no-fix poetry_docker_plugin tests
```
Comment (bruce-szalwinski-he) [inline Makefile:29]: added the `--diff` and `--no-fix` with the intention of using pre-commit to fix before commit and the github workflow to be a quality gate.

**Comment by bruce-szalwinski-he** [comment]
system tested with 

```
 poetry self add poetry-docker-plugin git+https://github.com/bruce-szalwinski-he/poetry-docker-plugin.git@multi-platform
 poetry docker --platform linux/amd64 --platform linux/arm64 --push
```

**Review by vagmcs — APPROVED** [review]
Everything seems fine. You can merge it. Please use squash and merge with a commit message starting with `bug: your message`, according to the [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0). This is helpful for me to generate changelogs for the new release from commit messages.

**Comment by bruce-szalwinski-he** [comment]
I don't have privileges to merge.  I updated the title of the PR which should become the commit message when you squash merge.  Conventional commit suggests using `fix` to patch a bug, so went with that.  Also, need to decide if you want the PR workflow (it is waiting for approval, could be updated to run on merges to main as well).  I can remove if you don't want it.

**Pull request merged**